### PR TITLE
autopilot: don't block Stop on pending connection

### PR DESCRIPTION
**Pull request by halseth**
To ensure a call to ConnectToPeer doesn't block the agent from
shutting down, we'll launch it in a non-waitgrouped goroutine, that
will signal when a result is returned.

This could cause `disabling` autopilot to hang if there were pending connections.

### CODE REVIEW

File: autopilot/agent_test.go [hunk 1]
```diff
@@ -1546,3 +1546,157 @@ func TestAgentSkipPendingConns(t *testing.T) {
...
+	select {
+	case <-connect:
+	case <-time.After(time.Second * 10):
+		t.Fatalf("agent did not attempt connection")
+	}
+
+	// Make sure that we are able to stop the agent, even though there is a
+	// pending connection.
+	stopped := make(chan error)
+	go func() {
```
Comment (cfromknecht) [inline autopilot/agent_test.go]: style nit: this can all be oneline :)

File: autopilot/agent_test.go [hunk 2]
```diff
@@ -1546,3 +1546,157 @@ func TestAgentSkipPendingConns(t *testing.T) {
...
+		t.Fatalf("unable to create agent: %v", err)
+	}
+
+	// To ensure the heuristic doesn't block on quitting the agent, we'll
+	// use the agent's quit chan to signal when it should also stop.
+	heuristic.quit = agent.quit
+
+	// With the autopilot agent and all its dependencies we'll start the
+	// primary controller goroutine.
+	// NOTE: We'll stop the agent manually below.
```
Comment (cfromknecht) [inline autopilot/agent_test.go]: should defer `Stop` anyway in case we fail before reaching the manual stop

File: autopilot/agent_test.go [hunk 3]
```diff
@@ -1546,3 +1546,157 @@ func TestAgentSkipPendingConns(t *testing.T) {
 		t.Fatalf("agent should have attempted connection")
 	}
 }
+
+// TestAgentQuitWhenPendingConns tests that we are able to stop the autopilot
+// agent even though there are pending connections to nodes.
+func TestAgentQuitWhenPendingConns(t *testing.T) {
```
Comment (wpaulino) [inline autopilot/agent_test.go]: Not a blocker, but looks like we'd really benefit from abstracting some common patterns within the tests into helper functions for easier readability.

File: autopilot/agent_test.go (hunk 2 above)
Comment (halseth) [reply autopilot/agent_test.go]: Fixed: https://github.com/lightningnetwork/lnd/pull/2550/commits/1b26041e2a18afabddafb7a3f858e6661be807bf

File: autopilot/agent_test.go (hunk 1 above)
Comment (halseth) [reply autopilot/agent_test.go]: Ah, like this? https://github.com/lightningnetwork/lnd/pull/2550/commits/eaf087ed89caac23fa0395863e3b7ab055c79e17

**Comment by halseth** [comment]
PTAL @cfromknecht @wpaulino

**Comment by wpaulino** [comment]
LGTM after squash.

**Comment by halseth** [comment]
Squashed and rebased.

**Pull request merged**